hash_practice/exercises.py

```diff
@@ -1,19 +1,35 @@
-
+import collections
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(N log N)
+        Space Complexity: O(N)
     """
-    pass
+    if strings == []:
+        return []
+
+    hash = collections.defaultdict(list)
+    for char in strings:

```

Comment:
`char` is usually short for "character" so this is a bit of a confusing name. `string` or `word` can work fine here.